### PR TITLE
Read cache-ispn.xml from conf/ by default

### LINKED ISSUE #31492: Misleading docs and functionality around cache-ispn.xml
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

The [docs](https://www.keycloak.org/server/caching) state: 

> Keycloak provides a cache configuration file with sensible defaults located at conf/cache-ispn.xml.

This could lead to the assumption that if this file is changed, the settings will come into effect (at least when `cache` is set to `ispn`). 

It also states:

>  You are free to change the number of owners accordingly to better fit into your availability requirements. To change the number of owners, open conf/cache-ispn.xml and change the value for owners=<value> for the distributed caches to your desired value.

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

When changing the file, those changes should come into effect once I edit the file (at least that's what the docs seem to tell me, and this is what I would assume when I see the file in the conf folder). 

### Actual behavior

The changes wouldn't come into effect unless I add the CLI parameter `--cache-config-file=cache-ispn.xml` as otherwise the file would be taken from the classpath which includes the shipped defaults.

### How to Reproduce?

This happend to me when I tried to enable the metrics in KC24 which need editing the `cache-ispn.xml`. A similar configuration mishap would happen if I change the number of cache entries. 

### Anything else?

While some of this could be fixed by the docs for KC24/KC25 where we don't want to change the functionality, keeping a file in the conf folder which contains the defaults but isn't used is part of the problem. 

Another idea: It might also be good in KC24/25 to print the file path where the ISPN file has been picked up from: 

https://github.com/keycloak/keycloak/blob/d2ae27a1e2ff0cf153f56e89c2dd3979eb6e7704/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/KeycloakRecorder.java#L143

This might have been discussed in the past, so I'm eager to hear the cloud native team's ideas around that.

### PR DESCRIPTION
**Pull request by pruivo**
Do not read the cache-ispn.xml shipped in the JAR since it cannot be changed

Fixed #31492

### CODE REVIEW

**Comment by pruivo** [comment]
cc @ahus1

**Comment by ahus1** [comment]
Thank you for this change. It makes sense, still we need to figure out if we can do it like this a minor release that it is supposed to be non-breaking. 

The only breaking changes are allowed are bug fixes as they are supposed to change the existing behavior. Still that is a fine line we shouldn't cross lightheartedly. 

Looking at the code, the cache file might now be required in more cases than before. Can you please describe how the behavior changes before and after your change?

**Comment by pruivo** [comment]
* Fixed the log message

**Review by vmuzikar — COMMENTED** [review]
@pruivo Thank you for the PR.

I believe this is not really a breaking change and could be merged as is. The only scenario where this would be a breaking change is if someone made some changes to their `conf/cache-ispn.xml` file but did not set the `--cache-config-file` option so effectively the file is use would still be the one from the JAR. But that I'd consider a misconfiguration.

Still, it'd be worth highlighting in the upgrading guide.

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/KeycloakRecorder.java
[the remark this answers is not in the document]
Comment (pruivo) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/KeycloakRecorder.java:1]: My bad!! Initially, I applied a fix in `KeycloakRecorder` but I changed my mind and fixed it in `CachingPropertyMapper`, where it makes more sense. I can revert the formatting changes and keep only the logger if you say so.

**Comment by pruivo** [comment]
* added behavior change into "changelog".

**Review by vmuzikar — DISMISSED** [review]

**Comment by pruivo** [comment]
* rebased to resolve the conflict. Please re-review, since I had to re-implement the logic in another place. Thanks!

**Review by ahus1 — APPROVED** [review]
Thank you for this fix!

**Review by vmuzikar — APPROVED** [review]

**Pull request merged**